### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kunyuan**
1. Add Vegas-like adaptive grid
2. Add benchmark example between other Vegas packages and this one. Conclusion: for low-dimensional integral, Vegas algorithm can be ~100 times fast, but for high-dimensional integral (D>4), our package becomes more robust and faster.

### CODE REVIEW

**Comment by iintSjds** [comment]
There are two binary files in cuba/ to be removed and gitignored

**Comment by kunyuan** [comment]
Fixed

File: src/adaptor/vegas.jl [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+
+""" 
+
+    function region(vegas::Vegas, d::Int=-1)
+
+x-space region. ``region(d)`` returns a tuple ``(xl,xu)`` specifying the ``x``-space
+interval covered by the map in direction ``d``. A list containing
+the intervals for each direction is returned if ``d`` is omitted.
+"""
+function region(vegas::Vegas, d::Int=-1)
```
Comment (iintSjds) [inline src/adaptor/vegas.jl:116]: will this cause type-instability?

**Comment by kunyuan** [comment]
This part is not done yet. I am translating that piece of code from python to Julia.

Best,
Kun

**Review by iintSjds — APPROVED** [review]

**Pull request merged**